docs/Using-Fleet/Osquery-process.md

```diff
@@ -0,0 +1,43 @@
+# Osquery children processes
+
+Osquery will run a watcher process to keep track of any child process and also any managed
+extensions. What follows is a description of what happens during the watcher REPL and under what
+circumstances the child process and/or managed extensions are terminated.
+
+As a first step, the watcher checks the state of the child worker process, which could be either `Alive` or
+`Non-existent`. If the process is `Alive`, we make sure the process is within its assigned resource
+quota, by checking: 
+
+1. That the maximum CPU utilization limit is not exceeded (which is controlled by osquery's 
+   `--watchdog_latency_limit` flag).
+
+2. The maximum memory limit is not exceeded (which is controlled by osquery's
+   `--watchdog_memory_limit` flag).
+	   
+If the child process is within the resource limits, then it is deemed alive and well, otherwise, we terminate the process by following these steps:
```

Comment:
```suggestion
If the child process is within the resource limits, then it is deemed alive and well. Otherwise, we terminate the process by following these steps:
```